app/src/main/java/net/programmierecke/radiodroid2/alarm/AlarmReceiver.java

```diff
@@ -215,8 +235,121 @@ protected void onPostExecute(String result) {
+
+        Handler handler = new Handler();
+        Runnable runnable = new Runnable() {
+            boolean isPlaying = false;
+            boolean hasAudioPlaybackStarted = false;
+            long playStartedMillis = 0;
+            float currentVolume = minVolume;
+            int i = 0;
+
+            @Override
+            public void run() {
+                i++;
+                if (BuildConfig.DEBUG) {
+                    Log.d(TAG, "Increase volume loop " + i);
+                }
+
+                long currentMillis = System.currentTimeMillis();
+                long totalElapsedMillis = currentMillis - triggerMillis;
+
+                try {
+                    isPlaying = itsPlayerService.isPlaying();
+                } catch (Exception ignore) {
+                }
+
+                if (playStartedMillis == 0 && isPlaying) {
+                    playStartedMillis = currentMillis;
+                    setAlarmVolume(minVolume, context);
+                }
+
+                try {
+                    hasAudioPlaybackStarted = itsPlayerService.hasPlaybackStarted();
+                } catch (Exception ignore) {
+                }
+
+                if (hasAudioPlaybackStarted) {
+                    long elapsedMillis = currentMillis - playStartedMillis;
+                    float slowWakeProgress = (float) (elapsedMillis / slowWakeMillis) / 10;
+
+                    if (currentVolume < maxVolume) {
+                        float newVolume = Math.min(slowWakeProgress, maxVolume);
+                        if (newVolume != currentVolume) {
+                            setAlarmVolume(newVolume, context);
+                            currentVolume = newVolume;
+                        }
+                        handler.postDelayed(this, 1000);
+                    } else {
+                        if (BuildConfig.DEBUG) {
+                            Log.d(TAG, "Volume 100% - stopping loop");
+                        }
+                        handler.removeCallbacks(this);
+                    }
+                } else if (!isPlaying && playStartedMillis > 0) {
+                    if (BuildConfig.DEBUG) {
+                        Log.d(TAG, "No longer playing - stopping loop");
+                    }
+                    handler.removeCallbacks(this);
+                } else if (totalElapsedMillis > streamReadTimeout) {
+                    if (BuildConfig.DEBUG) {
+                        Log.d(TAG, "Waiting for playback to start has exceeded streamReadTimeout - stopping");
+                    }
+                    try {
+                        itsPlayerService.Stop();
+                    } catch (Exception e) {
+                        if (BuildConfig.DEBUG) {
+                            Log.e(TAG, "Tried to stop stream, but failed with error:", e);
+                        }
+                    }
+                    PlaySystemAlarm(context);
+                    handler.removeCallbacks(this);
```

Comment:
You can wrap this part into a function and call if any try catch over service fails. I doubt there are cases where service fails and we shouldn't just start system sound. 

Or you can wrap the whole body of Runnable in try catch.